tests/scripts/all.sh

```diff
@@ -547,6 +547,12 @@ if_build_succeeded tests/ssl-opt.sh -f Default
 msg "test: compat.sh RC4, DES & NULL (full config)" # ~ 2 min
 if_build_succeeded env OPENSSL_CMD="$OPENSSL_LEGACY" GNUTLS_CLI="$GNUTLS_LEGACY_CLI" GNUTLS_SERV="$GNUTLS_LEGACY_SERV" tests/compat.sh -e '3DES\|DES-CBC3' -f 'NULL\|DES\|RC4\|ARCFOUR'
 
+msg "build: cmake, full config + DEPRECATED_REMOVED, clang, C99"
+# No cleanup: tweak the configuration, keep the makefiles
+scripts/config.pl set MBEDTLS_DEPRECATED_WARNING
```

Comment:
What's the purpose of having `MBEDTLS_DEPRECATED_WARNING` when `MBEDTLS_DEPRECATED_REMOVED` is set?

Reply:
They are used differently in the code. Should we do two builds, one with `MBEDTLS_DEPRECATED_WARNING` and one with `MBEDTLS_DEPRECATED_REMOVED`? A build is pretty cheap and we don't need to test.

Reply:
I've split the `DEPRECATED_WARNING` and `DEPRECATED_REMOVED` builds. I also added `-O`, which makes the build slower but is useful (it caught an issue in `cmac.c`) and wasn't done elsewhere. I don't see a need to cover `-O` and `DEPRECATED` separately: with `DEPRECATED_WARNING`, we're testing the same code.